.teamcity/settings.kts

```diff
@@ -53,22 +44,30 @@ project {
         }
     }
 
-    val deployConfigure = deployConfigure().apply {
+    val deployVersion = deployVersion().apply {
         dependsOnSnapshot(buildAll, onFailure = FailureAction.IGNORE)
+        dependsOnSnapshot(BUILD_CREATE_STAGING_REPO_ABSOLUTE_ID) {
+            reuseBuilds = ReuseBuilds.NO
+        }
     }
-    val deploys = platforms.map { deploy(it, deployConfigure) }
-    val deployPublish = deployPublish(deployConfigure).apply {
+    val deploys = platforms.map { deploy(it, deployVersion) }
+    val deployPublish = deployPublish(deployVersion).apply {
         dependsOnSnapshot(buildAll, onFailure = FailureAction.IGNORE)
+        dependsOnSnapshot(BUILD_CREATE_STAGING_REPO_ABSOLUTE_ID) {
+            reuseBuilds = ReuseBuilds.NO
+        }
         deploys.forEach {
             dependsOnSnapshot(it)
         }
     }
 
-    buildTypesOrder = listOf(buildAll, buildVersion, *builds.toTypedArray(), deployPublish, deployConfigure, *deploys.toTypedArray())
+    buildTypesOrder = listOf(buildAll, buildVersion, *builds.toTypedArray(), deployPublish, deployVersion, *deploys.toTypedArray())
+
+    additionalConfiguration()
```

Comment:
There probably should be `additionalConfiguration.kt` where this method is defined. Could you check that you have it committed?